### PR TITLE
remove `using namespace` from headers

### PR DESCRIPTION
**Pull request by makslevental**
Based on this recommendation https://github.com/Xilinx/mlir-aie/pull/719#discussion_r1382672981.

### CODE REVIEW

File: include/aie/Dialect/AIE/Transforms/AIEPathFinder.h [hunk 1]
```diff
@@ -140,21 +141,21 @@ class Pathfinder {
 } // namespace xilinx::AIE
 
 namespace llvm {
```
Comment (stephenneuendorffer) [inline include/aie/Dialect/AIE/Transforms/AIEPathFinder.h:143]: It's weird to add this to llvm::?

File: include/aie/Dialect/AIE/IR/AIE.td [hunk 2]
```diff
@@ -1117,19 +1117,12 @@ def AIE_LockOp: AIE_Op<"lock", [TileElement]>, Results<(outs Index)> {
...
     StringAttr name() {
-      if(auto attr = getOperation()->getAttrOfType<StringAttr>(SymbolTable::getSymbolAttrName())) {
+      if(auto attr = getOperation()->getAttrOfType<StringAttr>(SymbolTable::getSymbolAttrName()))
         return attr;
-      } else {
-        emitOpError("does not have '") << SymbolTable::getSymbolAttrName() <<
-          "' attribute specified";
-      }
+      emitOpError("does not have '") << SymbolTable::getSymbolAttrName() << "' attribute specified";
       llvm_unreachable("unreachable");
```
Comment (stephenneuendorffer) [inline include/aie/Dialect/AIE/IR/AIE.td:1126]: This is clearly reachable.  The llvm_unreachable is bogus.

File: include/aie/Dialect/AIE/IR/AIETargetModel.h [hunk 3]
```diff
@@ -463,29 +462,28 @@ class IPUTargetModel : public AIE2TargetModel {
   }
 };
 
-} // namespace AIE
-} // namespace xilinx
+} // namespace xilinx::AIE
 
 namespace llvm {
-using namespace xilinx::AIE;
```
Comment (stephenneuendorffer) [inline include/aie/Dialect/AIE/IR/AIETargetModel.h:470]: This one seems OK?

File: include/aie/Dialect/AIE/Transforms/AIEPathFinder.h [hunk 4]
```diff
@@ -140,21 +141,21 @@ class Pathfinder {
...
-using namespace xilinx::AIE;
```
Comment (stephenneuendorffer) [inline include/aie/Dialect/AIE/Transforms/AIEPathFinder.h:143]: OK?

**Review by stephenneuendorffer — APPROVED** [review]

File: include/aie/Dialect/AIE/IR/AIE.td (hunk 2 above)
Comment (makslevental) [reply include/aie/Dialect/AIE/IR/AIE.td:1126]: Got me - there's a compile error without this.

File: include/aie/Dialect/AIE/IR/AIETargetModel.h (hunk 3 above)
Comment (makslevental) [reply include/aie/Dialect/AIE/IR/AIETargetModel.h:470]: Yea I don't know but it doesn't smell right - adding a `using` for someone else's namespace in one of our namespaces seems fine - the reverse not so much.

File: include/aie/Dialect/AIE/Transforms/AIEPathFinder.h (hunk 1 above)
Comment (makslevental) [reply include/aie/Dialect/AIE/Transforms/AIEPathFinder.h:143]: What's weird? the `DenseMap` specialization or the `using`?

File: include/aie/Dialect/AIE/IR/AIE.td (hunk 2 above)
Comment (stephenneuendorffer) [reply include/aie/Dialect/AIE/IR/AIE.td:1126]: I think it should probably be an assert, not unreachable.

Comment (makslevental) [reply include/aie/Dialect/AIE/IR/AIE.td:1126]: Sorry I take my thumbs up back - how is that reachable if there's an `emitOpError`?

Comment (stephenneuendorffer) [reply include/aie/Dialect/AIE/IR/AIE.td:1126]: I think emitOpError prints a message, but doesn't fail.  in most circumstances, it's part of a transform where you return an expected, but obviously that can't happen in this situation.

File: include/aie/Dialect/AIE/IR/AIETargetModel.h (hunk 3 above)
Comment (stephenneuendorffer) [reply include/aie/Dialect/AIE/IR/AIETargetModel.h:470]: Yeah, playing further I think it's just not right, even within the namespace.

File: include/aie/Dialect/AIE/IR/AIE.td (hunk 2 above)
Comment (makslevental) [reply include/aie/Dialect/AIE/IR/AIE.td:1126]: Ah you're right - I was thinking of `report_fatal_error`.

**Pull request merged**